barretenberg/cpp/src/barretenberg/polynomials/polynomial.cpp

```diff
@@ -272,25 +275,34 @@ template <typename Fr> void Polynomial<Fr>::add_scaled(std::span<const Fr> other
     const size_t other_size = other.size();
     ASSERT(in_place_operation_viable(other_size));
 
-    /** TODO parallelize using some kind of generic evaluation domain
-     *  we really only need to know the thread size, but we don't need all the FFT roots
+    /** Calculates number of threads with calc_num_threads
+     *  Possible improvements: standardize this parallelization code
```

Comment:
Possible improvement done by your thread util? Our policy should be: either it's worth an explicit Github issue or it's not worth leaving as a TODO.

Reply:
You're welcome to go further and make a function that returns an array of three `size_t`s and then unpack as in
```
auto [num_threads, range_per_thread, leftovers] = get_parallel_for_parameters(...)
```

Reply:
thats a decent idea, but maybe I'll create a function that just does this whole thing, taking in number of iterations, and a function?